crates/turbopack-ecmascript/src/tree_shake/mod.rs

```diff
@@ -171,7 +178,12 @@ impl Analyzer<'_> {
                     if let Some(declarator) = &state.declarator {
                         if declarator != item_id {
                             // A write also depends on the declaration.
-                            self.g.add_weak_deps(item_id, [declarator].iter().copied());
+                            if item.side_effects {
+                                self.g
+                                    .add_strong_deps(item_id, [declarator].iter().copied());
+                            } else {
+                                self.g.add_weak_deps(item_id, [declarator].iter().copied());
+                            }
```

Comment:
The array should implement `IntoIterator<Item = &'a ItemId>`?

Apparently this was maybe an issue in earlier editions of rust prior to 2021? https://doc.rust-lang.org/std/primitive.array.html#editions

```suggestion
                                self.g
                                    .add_strong_deps(item_id, [declarator]);
                            } else {
                                self.g.add_weak_deps(item_id, [declarator]);
                            }
```